common/include/common/common_utils.hpp

```diff
@@ -1656,8 +1661,16 @@ template <typename scalar_t>
 static inline scalar_t random_scalar(scalar_t rangeMin, scalar_t rangeMax) {
   static std::random_device rd;
   static std::default_random_engine gen(rd());
-  std::uniform_real_distribution<scalar_t> dis(rangeMin, rangeMax);
-  return dis(gen);
+  if constexpr (std::is_same_v<scalar_t, cl::sycl::half>) {
```

Comment:
I think that this function needs to be consistent with the `test/blas_test.hpp` one.

Reply:
yes will update it accordingly.